File: Cubism3-ARKit/MenuCell.m
//
//  MenuCell.m
//  Cubism3-ARKit
//
//  Created by Yuma Matsune on 2018/08/18.
//  Copyright © 2018 matsune. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "MenuCell.h"

@interface MenuCell()

@property (weak, nonatomic) IBOutlet UIImageView *imageView;
@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation MenuCell

- (void)bindImage:(UIImage*)image Title:(NSString*)title {
    [self.imageView setImage:image];
    [self.label setText:title];
}

@end

File: Cubism3-ARKit/menu/MenuCell.swift
//
//  MenuCell.swift
//  Cubism3-ARKit
//
//  Created by Yuma Matsune on 2018/08/18.
//  Copyright © 2018 matsune. All rights reserved.
//

import Foundation
import UIKit

final class MenuCell: UICollectionViewCell {

    @IBOutlet weak var imageView: UIImageView!
    @IBOutlet weak var label: UILabel!
    
    override func prepareForReuse() {
        super.prepareForReuse()
        isSelected = false
    }

    override var isSelected: Bool {
        didSet {
            if isSelected {
                backgroundColor = .lightGray
            } else {
                backgroundColor = .clear
            }
        }
    }
    
    func bind(menu: MenuItem) {
        imageView.image = menu.image
        label.text = menu.title
    }
}
